CHANGELOG.md

```diff
@@ -458,6 +458,21 @@ Additions to existing modules
   product-↭ : product Preserves _↭_ ⟶ _≡_
   ```
 
+* Added some very-dependent map and zipWith to `Data.Product`.
+  ```agda
+  dep-map : {B : A → Set b} {P : A → Set p} {Q : {x : A} → P x → B x → Set q} →
+   (f : (x : A) → B x) → (∀ {x} → (y : P x) → Q y (f x)) →
+   (v : Σ A P) → Σ (B (proj₁ v)) (Q (proj₂ v))
```

Comment:
I have a personal aversion to `proj₁` and `proj₂` occurring in types like this, so a possible alternative is to use an irrefutable `@`-pattern in the type, i.e.
```agda
(v@(x , y) : Σ A P) → Σ (B x) (Q y)
```
etc.

Reply:
Apologies! Agda allows this in `module` telescopes (I think!?), but not in ordinary function telescopes... so you'd need 
```agda
   (v : Σ A P) → (let x , y = v) → Σ (B x) (Q y)
```
instead...
... but still worth it, I think!

Reply:
Ha! Fixed in 2.6.4.*, but fails in v2.6.3... version error on my part!

Reply:
or even (more) simply:
```agda
((x , y) : Σ A P) → Σ (B x) (Q y)
```

Reply:
Naming : `map-Σ`? or `Σ-map`?

Reply:
All good suggestions, all done.